load/README.md

```diff
@@ -0,0 +1,130 @@
+# Load
+
+This package provides generic utilities for blockchain load testing. We break load generation down into the following components:
+
+- tx generator(s)
+- tx issuer(s)
+- tracker
+- orchestrator
+
+The transaction generator(s) and issuer(s) may be VM specific and provide the
+necessary injected dependencies for the orchestrator. This enables us to
+construct different load testing strategies on top of the same re-usable code.
+For example, we can re-use these components for a short burst of transactions or
+to perform gradual load testing.
+
+## Architecture
+
+```mermaid
+graph
+    A["Orchestrator"]
+    subgraph B1["Issuer 1"]
+        TG1["TxGenerator 1"]
+    end
+    subgraph B2["Issuer 2"]
+        TG2["TxGenerator 2"]
+    end
+    subgraph B3["Issuer 3"]
+        TG3["TxGenerator 3"]
+    end
+    C["Tracker"]
+
+    A <--> C
+
+    A --> B1
+    A --> B2
+    A --> B3
+
+    B1 --> C
+    B2 --> C
+    B3 --> C
+```
+
+### Orchestrator
+
+The orchestrator is responsible for directing issuers to send transactions to
+the network. The strategy for how the orchestrator directs issuers varies
+between implementations (e.g. short burst vs gradual load).
+
+When applicable, the orchestrator can also query the tracker to make a decision
+(e.g. increase TPS or return).
+
+### Transaction Generator
+
+The TX generator is responsible for generating a valid transaction which any
+issuer can send. Each TX generator is assumed to have an associated account that
+will sign transactions and whose balance is monitored to err in the case of fund exhaustion.
+
+### TX Issuer
+
+The TX issuer has two responsibilities:
+- Sending transactions to the network
+- Listening for the status of transactions it has sent
+
+The TX issuer also has the following obligations to the tracker:
+
+- Upon sending a TX, the issuer should call `Issue()`
+- Upon confirming an accepted TX, the issuer should call `ObserveConfirmed()`
+- Upon confirming a failed TX, the issuer should call `ObserveFailed()`
+
```

Comment:
```suggestion
The Tx issuer is responsible for sending and confirming transactions. As it issues/confirms transactions, it notifies the tracker, so that it can maintain metrics across the generated load.
```